<a href="https://www.kaggle.com/code/saadatkhalid/eda-number-of-visitors-to-japan-codanics?scriptVersionId=150403491" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

<div style="background-color: #FF7F50; color: #FFFFFF; padding: 10px; font-size: 24px; font-weight: bold; text-align: center; border-radius: 5px;">Project Title: EDA - Number of visitors to Japan</div>

# About Dataset
year: 2023\
data: government statistics\
__Dataset is taken from [Kaggle](https://www.kaggle.com/datasets/risakashiwabara/japannumber-of-visitors-to-japan/data)__

___


### Author: *Saadat Khalid Awan*
**Email:** *me.saadi96@gmail.com*\
**Website:** *https://thesaadat.blogspot.com/*

## 🌐 Let's Connect:
[![Facebook](https://img.shields.io/badge/Facebook-%231877F2.svg?logo=Facebook&logoColor=white)](https://facebook.com/Saadat.Khalid.Awan) [![Instagram](https://img.shields.io/badge/Instagram-%23E4405F.svg?logo=Instagram&logoColor=white)](https://instagram.com/saadii_awan66) [![LinkedIn](https://img.shields.io/badge/LinkedIn-%230077B5.svg?logo=linkedin&logoColor=white)](https://linkedin.com/in/saadatawan) [![Medium](https://img.shields.io/badge/Medium-12100E?logo=medium&logoColor=white)](https://medium.com/@@me.saadat) [![Pinterest](https://img.shields.io/badge/Pinterest-%23E60023.svg?logo=Pinterest&logoColor=white)](https://pinterest.com/its_saadatkhalid) [![Quora](https://img.shields.io/badge/Quora-%23B92B27.svg?logo=Quora&logoColor=white)](https://quora.com/profile/Saadat-Khalid-Awan) [![TikTok](https://img.shields.io/badge/TikTok-%23000000.svg?logo=TikTok&logoColor=white)](https://tiktok.com/@saadat.awan) [![Twitter](https://img.shields.io/badge/Twitter-%231DA1F2.svg?logo=Twitter&logoColor=white)](https://twitter.com/saadat_96) [![YouTube](https://img.shields.io/badge/YouTube-%23FF0000.svg?logo=YouTube&logoColor=white)](https://youtube.com/@saadatkhalidawan) [![Github](https://img.shields.io/badge/Github-%23FF0000.svg?logo=Github&logoColor=Black)](https://github.com/Saadat-Khalid/)


# 1. Importing Libraries

In [1]:
# libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

### Important Step: To display all values in a pandas DataFrame

# 2. Data Loading


 ↪ Load the csv file with the pandas
 
 ↪ creating the dataframe and understanding the data present in the dataset using pandas



__There are two `datasets` one is `"Number of foreign visitors to Japan by month"` and other one is `"Number of foreign visitors to Japan by year"`__ 

- First I'll Check the `by month` and after `by year`


### Number of foreign visitors to Japan by month

In [2]:
df = pd.read_csv('/kaggle/input/japannumber-of-visitors-to-japan/Number of foreign visitors to Japan by month_ .csv')

- Viewing the first five Rows of the data

In [3]:
df.head()

Country/Area Month   Year  Visitor
0  Afghanistan   Jan.  2017      233
1  Afghanistan   Feb.  2017      196
2  Afghanistan   Mar.  2017      192
3  Afghanistan   Apr.  2017      259
4  Afghanistan    May  2017      228

- Viewing the Columns present in the dataset

In [4]:
df.columns

Index(['Country/Area', 'Month ', 'Year', 'Visitor'], dtype='object')

* There are 4 columns in this dataset
  * `Country/Area`, `Month`, `Year`, `Visitor`

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17052 entries, 0 to 17051
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Country/Area  17052 non-null  object
 1   Month         17052 non-null  object
 2   Year          17052 non-null  int64 
 3   Visitor       17052 non-null  int64 
dtypes: int64(2), object(2)
memory usage: 533.0+ KB


- The total number of `Rows` are 17052
- The total number of `Columns` are 4
- `Country/Area` and `Month` columns are `object` dtype
- `Year` and `Visitors` columns are `int64` dtype

### Number of foreign visitors to Japan by Year

In [6]:
df_year = pd.read_csv('/kaggle/input/japannumber-of-visitors-to-japan/Number of foreign visitors to Japan by year_ .csv')

- Viewing the first five Rows of the data

In [7]:
df_year.head()

Year  Visitor Arrivals
0  1964            352832
1  1965            366649
2  1966            432937
3  1967            476771
4  1968            519004

- Viewing the Columns present in the dataset

In [8]:
df_year.columns

Index(['Year', 'Visitor Arrivals'], dtype='object')

* There are 2 columns in this dataset
  * `Year`, `Visitors Arrivals`

In [9]:
df_year.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60 entries, 0 to 59
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   Year              60 non-null     int64
 1   Visitor Arrivals  60 non-null     int64
dtypes: int64(2)
memory usage: 1.1 KB


- The total number of `Rows` are 60
- The total number of `Columns` are 2
- `Year` and `Visitor Arrivals` columns are `int64` dtype

### Checking for Null/Missing Values in the both Datasets

In [10]:
df.isnull().sum() # for vistors by months

Country/Area    0
Month           0
Year            0
Visitor         0
dtype: int64

In [11]:
df_year.isnull().sum() # for vistors by year

Year                0
Visitor Arrivals    0
dtype: int64

- The Dataset contains no missing/null values.

## How many Countries are there in the dataset?

In [12]:
df['Country/Area'].nunique()

203

In [13]:
df['Country/Area'].value_counts().sort_values(ascending=True)

Country/Area
Afghanistan       84
Brunei            84
Algeria           84
Andorra           84
Angola            84
                  ..
Venezuela         84
Vietnam           84
Yemen             84
United Kingdom    84
Zimbabwe          84
Name: count, Length: 203, dtype: int64

## Obervation: 
___
 
There are total `203` countries in the dataset. Where each country count is `84`. If we multiple `84*203` we get total number of rows `17,052`

## Which Country visits to Japan more each Year?

In [14]:
df.columns

Index(['Country/Area', 'Month ', 'Year', 'Visitor'], dtype='object')

In [15]:
top_countries_by_year = df.groupby(['Year', 'Country/Area'])['Visitor']
top_countries_by_year.sum().groupby('Year').nlargest(5)

Year  Year  Country/Area 
2017  2017  China            7355818
            South Korea      7140438
            Taiwan           4564053
            Hong Kong        2231568
            United States    1374964
2018  2018  China            8380034
            South Korea      7538952
            Taiwan           4757258
            Hong Kong        2207804
            United States    1526407
2019  2019  China            9594394
            South Korea      5584597
            Taiwan           4890602
            Hong Kong        2290792
            United States    1723861
2020  2020  China            1069256
            Taiwan            694476
            South Korea       487939
            Hong Kong         346020
            Thailand          219830
2021  2021  China              42239
            Vietnam            26586
            United States      20026
            South Korea        18947
            Nepal              11648
2022  2022  South Korea      1012751
            

In [16]:
top_countries_by_year = df.groupby(['Year', 'Country/Area'])['Visitor'].sum().reset_index()
top_countries_by_year = top_countries_by_year.groupby('Year').apply(lambda x: x.nlargest(10, 'Visitor')).reset_index(drop=True)

fig = px.bar(top_countries_by_year, x='Year', y='Visitor', color='Country/Area', title='Top 10 Countries by Visitor Count')
fig.show()

## Observation:
___

- China has consistently been the largest source of visitors to Japan from 2017 to 2023, except for a significant decrease in 2021.
- South Korea and Taiwan are the second and third most common countries of origin for visitors to Japan, following China.
- Hong Kong and the United States are also among the top five countries of origin for visitors to Japan.
- The number of visitors from South Korea to Japan has shown a significant increase over the years.
- Taiwan has had a stable number of visitors to Japan, with a slight increase in 2023.
- The United States has experienced a gradual increase in visitor count to Japan, reaching its peak in 2023.
- Hong Kong has shown fluctuating visitor numbers, with a notable decrease in 2023.
- Other countries such as Thailand, Vietnam, and Nepal also contribute to the visitor count to Japan, but in smaller numbers compared to the top countries mentioned above.


## Which Country Least Visits to Japan each Year?

In [17]:
lowest_countries_by_year = df.groupby(['Year', 'Country/Area'])['Visitor']
lowest_countries_by_year.sum().groupby('Year').nsmallest(10)

Year  Year  Country/Area         
2017  2017  East Germany             0
            Serbia and Montenegro    0
            South Yemen              0
            Unknown                  0
            Central Africa           5
                                    ..
2023  2023  Sao Tome and Principe    1
            Equatorial Guinea        2
            Vatican                  2
            Guinea-Bissau            6
            Central Africa           8
Name: Visitor, Length: 70, dtype: int64

In [18]:
lowest_countries_by_year = df.groupby(['Year', 'Country/Area'])['Visitor'].sum().reset_index()
lowest_countries_by_year = lowest_countries_by_year.groupby('Year').apply(lambda x: x.nsmallest(10, 'Visitor')).reset_index(drop=True)

fig = px.bar(lowest_countries_by_year, x='Year', y='Visitor', color='Country/Area', title='Least 10 Countries by Visitor Count')
fig.show()

## Observation:
____

- East Germany, Serbia and Montenegro, South Yemen, and Unknown had no recorded visitors to Japan from 2017 to 2023, indicating minimal or no tourist exchange with Japan during this period.
- Central Africa, Equatorial Guinea, Sao Tome and Principe, and Vatican had consistently low visitor numbers to Japan, suggesting a relatively small number of tourists from these countries.
- Guinea-Bissau showed a small but increasing interest in visiting Japan, with visitor counts ranging from 13 in 2017 to 30 in 2019 and 6 in 2023.
- North Korea had a small number of visitors to Japan, with counts ranging from 13 in 2018 and 2019 to 0 in subsequent years, suggesting potential travel restrictions or limitations.
- Saint Vincent, Micronesia, Belize, Nauru, San Marino, and Saint Kitts and Nevis recorded a small number of visitors to Japan in specific years, indicating limited tourist exchange with Japan.


## How many Visitors Arrival Over Year?

In [19]:
df_year.groupby('Year')['Visitor Arrivals'].sum()

Year
1964      352832
1965      366649
1966      432937
1967      476771
1968      519004
1969      608744
1970      854419
1971      660715
1972      723744
1973      784691
1974      764246
1975      811672
1976      914772
1977     1028140
1978     1038875
1979     1112606
1980     1316632
1981     1583043
1982     1793164
1983     1968461
1984     2110346
1985     2327047
1986     2061526
1987     2154864
1988     2355412
1989     2835064
1990     3235860
1991     3532651
1992     3581540
1993     3410447
1994     3468055
1995     3345274
1996     3837113
1997     4218208
1998     4106057
1999     4437863
2000     4757146
2001     4771555
2002     5238963
2003     5211725
2004     6137905
2005     6727926
2006     7334077
2007     8346969
2008     8350835
2009     6789658
2010     8611175
2011     6218752
2012     8358105
2013    10363918
2014    13413473
2015    19737409
2016    24039700
2017    28691073
2018    31191856
2019    31882049
2020     4115828
2021      245862
2022     

In [20]:
fig = px.scatter(df_year, x='Year', y='Visitor Arrivals', size = 'Visitor Arrivals', color='Visitor Arrivals' ,title='Visitor Arrivals to Japan Over Time')
fig.update_layout(xaxis_title='Year', yaxis_title='Visitor Arrivals')

# Show the plot
fig.show()

## Observation:
___


- The number of visitor arrivals to Japan has generally increased over the years.
- There was a significant increase starting from the 1970s, with a peak of 24 million arrivals in 2016.
- 2017 saw a further increase to nearly 29 million arrivals.
- In 2018 and 2019, there were over 31 million arrivals each year.
- However, due to the COVID-19 pandemic, the number dropped to around 4 million in 2020.
- In 2021, there were only approximately 246,000 arrivals.
- The years 2022 and 2023 show signs of recovery with 3.8 million and 10.7 million arrivals, respectively.


### The END

____
### Author: *Saadat Khalid Awan*
**Email:** *me.saadi96@gmail.com*\
**Website:** *https://thesaadat.blogspot.com/*

## 🌐 Let's Connect:
[![Facebook](https://img.shields.io/badge/Facebook-%231877F2.svg?logo=Facebook&logoColor=white)](https://facebook.com/Saadat.Khalid.Awan) [![Instagram](https://img.shields.io/badge/Instagram-%23E4405F.svg?logo=Instagram&logoColor=white)](https://instagram.com/saadii_awan66) [![LinkedIn](https://img.shields.io/badge/LinkedIn-%230077B5.svg?logo=linkedin&logoColor=white)](https://linkedin.com/in/saadatawan) [![Medium](https://img.shields.io/badge/Medium-12100E?logo=medium&logoColor=white)](https://medium.com/@@me.saadat) [![Pinterest](https://img.shields.io/badge/Pinterest-%23E60023.svg?logo=Pinterest&logoColor=white)](https://pinterest.com/its_saadatkhalid) [![Quora](https://img.shields.io/badge/Quora-%23B92B27.svg?logo=Quora&logoColor=white)](https://quora.com/profile/Saadat-Khalid-Awan) [![TikTok](https://img.shields.io/badge/TikTok-%23000000.svg?logo=TikTok&logoColor=white)](https://tiktok.com/@saadat.awan) [![Twitter](https://img.shields.io/badge/Twitter-%231DA1F2.svg?logo=Twitter&logoColor=white)](https://twitter.com/saadat_96) [![YouTube](https://img.shields.io/badge/YouTube-%23FF0000.svg?logo=YouTube&logoColor=white)](https://youtube.com/@saadatkhalidawan) [![Github](https://img.shields.io/badge/Github-%23FF0000.svg?logo=Github&logoColor=Black)](https://github.com/Saadat-Khalid/)


_____